modules/tinymce/src/core/main/ts/NodeChange.ts

```diff
@@ -97,7 +97,7 @@ class NodeChange {
     let node: Element | undefined;
 
     // Fix for bug #1896577 it seems that this can not be fired while the editor is loading
-    if (this.editor.initialized && selection && !Options.shouldDisableNodeChange(this.editor) && !this.editor.mode.isReadOnly()) {
+    if ((this.editor.initialized && selection && !Options.shouldDisableNodeChange(this.editor)) || (this.editor.mode.isReadOnly() && this.editor.mode.isSelectionEnabled())) {
```

Comment:
Maybe I am reading the `if` block incorrectly but it looks like `(this.editor.mode.isReadOnly() && this.editor.mode.isSelectionEnabled())` can override the previous `(this.editor.initialized && selection && !Options.shouldDisableNodeChange(this.editor))` check. Is this intentional?

Reply:
I think we should also still respect if `shoulDisableNodeChange` option if the integrator decide to switch that off, I'll push a change to fix that.

Reply:
The change looks better 👍